pkg/naming/metric_namer.go

```diff
@@ -102,10 +102,15 @@ type metricNamer struct {
 	nameMatches    *regexp.Regexp
 	nameAs         string
 	seriesMatchers []*ReMatcher
+	namespaced     *bool
 
 	ResourceConverter
 }
 
+func (n *metricNamer) isNamespaced() bool {
+	return (n.namespaced != nil) && *n.namespaced
```

Comment:
IMO, we should still enforce the `namespace` label by default and only drop the enforcement if `namespaced: false`.